lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -1039,55 +1079,153 @@ def configure_cpu_and_memory(self, vm_obj, vm_creation=False):
                 self.configspec.firmware = boot_firmware
                 self.change_detected = True
 
+    def get_configured_cdroms(self):
+        # cdroms {'ide': [{num: 0, cdrom: []}, {}], 'sata': [{num: 0, cdrom: []}, {}, ...]}
+        cdroms = {'ide': None, 'sata': None}
+        for cdrom_spec in self.params.get('cdrom'):
+            if 'type' not in cdrom_spec or cdrom_spec['type'] not in ['none', 'client', 'iso']:
+                self.module.fail_json(msg="cdrom.type is mandatory. Options are 'none', 'client', and 'iso'.")
+            if cdrom_spec['type'] == 'iso' and ('iso_path' not in cdrom_spec or not cdrom_spec['iso_path']):
+                self.module.fail_json(msg="cdrom.iso_path is mandatory when cdrom.type is set to iso.")
+            if 'controller_type' in cdrom_spec and cdrom_spec['controller_type'].lower() in ['sata', 'ide']:
+                cdrom_ctl = cdrom_spec['controller_type'].lower()
```

Comment:
Just hardcode the value: `cdrom_ctl = 'ide'`

Reply:
I'll add SATA support later after this PR and another multiple disk controllers and disks PR merged.